src/network.cpp

```diff
@@ -68,26 +72,45 @@ void setupWebServer() {
   });
   webServer.on("/wifi", HTTP_ANY, [](AsyncWebServerRequest *request) {
     switch (request->method()) {
-      case HTTP_GET:
-        request->send_P(200, "text/html", WIFI_HTML_PROGMEM_ARRAY,
-                        WIFI_HTML_SIZE, templateProcessor);
+    case HTTP_GET:
+      request->send_P(200, "text/html", WIFI_HTML_PROGMEM_ARRAY, WIFI_HTML_SIZE,
+                      templateProcessor);
+      return;
+    case HTTP_POST:
+      const auto ssidParam = request->getParam("s", true);
+      const auto passwordParam = request->getParam("p", true);
+      if (ssidParam == nullptr || passwordParam == nullptr ||
+          ssidParam->value().length() > sizeof(Settings.ap_name) ||
+          passwordParam->value().length() > sizeof(Settings.ap_password)) {
+        request->send(200, "text/html", "Invalid SSID or Password.");
```

Comment:
401 error ?

Reply:
This should also be a 400 error, not 401.